lib/memory_vty.c

```diff
@@ -28,7 +28,9 @@
 #include <malloc/malloc.h>
 #endif
 #include <dlfcn.h>
+#ifdef HAVE_DLINFO_LINKMAP
```

Comment:
This should be `#ifdef HAVE_LINK_H`

Reply:
I put HAVE_DLINFO_LINKMAP as that we are only in that esction of code below, but I can change it.

Reply:
I know that's what it's used by, but that might not be the case forever, and also I just cleaned up other pieces like this in #2965 (e.g. ade3cc5d298413b336eceb9ca7614b26f1fab9a9 and a6ff6159be10247a6e99bc0ba7011bf237853e41) so I don't want to add new semi-correct ones :grin:

Reply:
fixed